OPES-1588: Add some documentation. (#226)
openeuropa/oe_authentication: README.md

See the [Cas module](https://www.drupal.org/project/cas) for more information.

### Proxy

You can configure the module to "Initialize this client as a proxy" which allows
authentication requests to 3rd party services (e.g. ePOETRY).

```php
$config['cas.settings']['proxy']['initialize'] = TRUE;
```

See the [Cas module](https://www.drupal.org/project/cas) for more information.

### Two-factor authentication
The module allows to configure if two-factor authentication (in short, 2FA) is required for users to login.
It supports three modes:
* _Never_: 2FA is never required for any registered user account.
* _Always_: 2FA is always required for any registered user account. A 2FA authentication method will be enforced directly on EU Login.

> * _Based on conditions_: 2FA will be required only for user accounts that match one of the configured conditions. See below.

Reviewer: ```suggestion
* _Based on conditions_: 2FA is required only for registered user accounts that matches one of the configured conditions. See below.
```

Reviewer: match not matches

**Two-factor authentication conditions**

The conditions are validated after the user logged in via EU Login, and only if
the login attempt was executed without using a 2FA authentication method.\

> If at least one of the configured condition matches the user account, the login

Reviewer: ```suggestion
If at least one of the configured condition matches the user account and it is not meet, the login
```

Author: Actually the condition has to match, so adding that is contradictory. I will rephrase.

attempt will be rejected.\

> A message will be shown to the user to re-login using a suitable method, together

Reviewer: ```suggestion
A message will be shown to the user to try again using a suitable method, together
```
